napari/plugins/io.py

```diff
@@ -132,15 +133,25 @@ def save_layers(
         File paths of any files that were written.
     """
     if len(layers) > 1:
-        return _write_multiple_layers_with_plugins(
+        written = _write_multiple_layers_with_plugins(
             path, layers, plugin_name=plugin
         )
-    if len(layers) == 1:
+    elif len(layers) == 1:
         written = _write_single_layer_with_plugins(
             path, layers[0], plugin_name=plugin
         )
-        return [written] if written else []
-    return []
+        written = [written] if written else []
+
+    if not written:
```

Comment:
this will be an NameError if len(layers) == 0

Reply:
thank you!  fixed, and added a test